### PR TITLE
feat: always use markdown for the documentation

### PR DESCRIPTION
**Pull request by alacambre-yseop**
By default, we were using String to represent documentation during completion. Now it is treated as a markdown String, allowing a better display in suggestion list.

### CODE REVIEW

File: server/src/test/YmlKaoFileVisitor.test.ts [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
+import { CompletionItemKind, MarkupContent } from 'vscode-languageserver';
 
 import { YmlCompletionItemsProvider } from '../completion/YmlCompletionItemsProvider';
 import { YmlDefinitionProvider } from '../definitions';
 import { YmlLexer, YmlParser } from '../grammar';
 import { YmlKaoFileVisitor } from '../visitors';
 
+const NOT_DOCUMENTED: MarkupContent = {
+    kind: 'markdown',
+    value: 'not documented',
```
Comment (mminot-yseop) [inline server/src/test/YmlKaoFileVisitor.test.ts]:
```suggestion
    value: 'Not documented.',
```

File: server/src/yml-objects/AbstractYmlObject.ts [hunk 2]
```diff
@@ -81,14 +84,30 @@ export abstract class AbstractYmlObject implements CompletionItem {
...
         this.detail = getType(fields, connection, baseType);
         if (scopeEndOffset && scopeStartOffset) {
             this.scopeEndOffset = scopeEndOffset;
             this.scopeStartOffset = scopeStartOffset;
         }
     }
 
+    /**
+     * Set the documentation of this object as a `MarkupContent`.
+     * @param doc The documentation as string of this object.
```
Comment (mminot-yseop) [inline server/src/yml-objects/AbstractYmlObject.ts]:
```suggestion
     * @param doc The documentation of this object, as a Markdown string.
```

File: server/src/yml-objects/AbstractYmlObject.ts [hunk 3]
```diff
@@ -81,14 +84,30 @@ export abstract class AbstractYmlObject implements CompletionItem {
...
+        this.setDocumentation(getDocumentation(fields, connection));
         this.detail = getType(fields, connection, baseType);
         if (scopeEndOffset && scopeStartOffset) {
             this.scopeEndOffset = scopeEndOffset;
             this.scopeStartOffset = scopeStartOffset;
         }
     }
 
+    /**
+     * Set the documentation of this object as a `MarkupContent`.
```
Comment (mminot-yseop) [inline server/src/yml-objects/AbstractYmlObject.ts]:
```suggestion
     * Set the documentation of this object as a `MarkupContent`.
     * Set the documentation to `null` if empty.
```

File: .vscode/settings.json [hunk 4]
```diff
@@ -0,0 +1 @@
+{}
```
[the remark this answers is not in the document]
Comment (alacambre-yseop) [reply .vscode/settings.json]: Generated by VSCode itself. I suppose this can't be a bad thing to have it.

**Pull request merged**